src/convertToObject.js

```diff
@@ -7,6 +7,21 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  return sourceString
+    .split(';')
+    .map((item) => item.trim())
+    .filter((item) => item.includes(':'))
+    .reduce((obj, item) => {
+      const [property, ...value] = item.split(':');
+      const trimProperty = property.trim();
+      const trimValue = value.join(':').trim();
+
+      if (trimProperty && trimValue) {
```

Comment:
The check for `trimProperty` and `trimValue` before assignment is a good practice to avoid adding undefined or empty keys and values to the result object.